src/scripts/farming/Plot.ts

```diff
@@ -95,6 +97,28 @@ class Plot implements Saveable {
             return GameConstants.formatTime(timeLeft / growthMultiplier);
         });
 
+        this.formattedBaseStageTimeLeft = ko.pureComputed(() => {
+            if (this.berry === BerryType.None) {
+                return '';
+            }
+            const growthTime = this.berryData.growthTime.find(t => this.age < t);
+            const timeLeft = Math.ceil(growthTime - this.age);
+            return GameConstants.formatTime(timeLeft / 1);
+        });
+
+        this.formattedBaseTimeLeft = ko.pureComputed(() => {
+            if (this.berry === BerryType.None) {
+                return '';
+            }
+            let timeLeft = 0;
+            if (this.age < this.berryData.growthTime[3]) {
+                timeLeft = Math.ceil(this.berryData.growthTime[3] - this.age);
+            } else {
+                timeLeft = Math.ceil(this.berryData.growthTime[4] - this.age);
+            }
+            return GameConstants.formatTime(timeLeft / 1);
+        });
```

Comment:
Can you refactor these with the regular ones above so we don't have duplicated code?
For instance, you could have a `calcFormattedTimeLeft = (includeGrowthMultiplier: boolean) => {` which can be called in the regular and base computeds with true/false depending on if they want to use the growth multiplier or not. Just saves us having to modify two places in the future if anything changes.

Reply:
>`calcFormattedTimeLeft = (includeGrowthMultiplier: boolean) => {`

Would that be instead of my added calculation on line 100 or the original on line 86? And how would I go about formatting a boolean setting? I'm having trouble visualizing how I could include the data and accompanying text in the same function and result line, if that's what you're describing.

Reply:
There are currently these two functions in the code with your changes
```ts
        this.formattedTimeLeft = ko.pureComputed(() => {
            if (this.berry === BerryType.None) {
                return '';
            }
            let timeLeft = 0;
            if (this.age < this.berryData.growthTime[3]) {
                timeLeft = Math.ceil(this.berryData.growthTime[3] - this.age);
            } else {
                timeLeft = Math.ceil(this.berryData.growthTime[4] - this.age);
            }
            const growthMultiplier = App.game.farming.getGrowthMultiplier() * this.getGrowthMultiplier();
            return GameConstants.formatTime(timeLeft / growthMultiplier);
        });

        this.formattedBaseTimeLeft = ko.pureComputed(() => {
            if (this.berry === BerryType.None) {
                return '';
            }
            let timeLeft = 0;
            if (this.age < this.berryData.growthTime[3]) {
                timeLeft = Math.ceil(this.berryData.growthTime[3] - this.age);
            } else {
                timeLeft = Math.ceil(this.berryData.growthTime[4] - this.age);
            }
            return GameConstants.formatTime(timeLeft / 1);
        });
```

They are identical except for the one line defining a growthMultiplier, and the last line where one uses and the other just hardcodes a value of 1.

Instead, you should do something like this
```ts
        this.calcFormattedTimeLeft = (includeGrowthMultiplier: boolean) => {
            if (this.berry === BerryType.None) {
                return '';
            }
            let timeLeft = 0;
            if (this.age < this.berryData.growthTime[3]) {
                timeLeft = Math.ceil(this.berryData.growthTime[3] - this.age);
            } else {
                timeLeft = Math.ceil(this.berryData.growthTime[4] - this.age);
            }
            const growthMultiplier = includeGrowthMultiplier
                ? App.game.farming.getGrowthMultiplier() * this.getGrowthMultiplier()
                : 1;
            return GameConstants.formatTime(timeLeft / growthMultiplier);
        });

        this.formattedTimeLeft = ko.pureComputed(() => {
            return this.calcFormattedTimeLeft(true);
        });

        this.formattedBaseTimeLeft = ko.pureComputed(() => {
            return this.calcFormattedTimeLeft(false);
        });
```

and something similar for the StageTimeLeft functions.